src/main/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrdering.java

```diff
@@ -0,0 +1,149 @@
+            optional = true)
+    public boolean skipCommandLinePrompt = false;
+
+    private VariantContextWriter writer;
+
+    @Override
+    public void onTraversalStart() {
+        assertThatTheyReallyWantToProceed();
+
+        if (batchSize == 0) {
+            throw new UserException("your callset is not affected by the bug if you ran with "+ GenomicsDBImport.BATCHSIZE_ARG_NAME +" 0");
+        }
+
+        final VCFHeader originalHeader = getHeaderForVariants();
+        final Set<VCFHeaderLine> originalHeaderLines = originalHeader.getMetaDataInInputOrder();
+        final Set<VCFHeaderLine> newHeaderLines = new LinkedHashSet<>(originalHeaderLines);
+        newHeaderLines.addAll(getDefaultToolVCFHeaderLines());
+
+        final List<String> sampleNamesOriginalOrdering = getSampleNamesInInputOrder(sampleNameMapPath);
+        if( sampleNamesOriginalOrdering.size() <= batchSize ){
+            throw new UserException("you are not affected by the sample name ordering bug if your batch size is >= the number of samples in your callset. \n"
+                                            + "batch size: " + batchSize + "\n"
+                                            + "number of samples: " + sampleNamesOriginalOrdering.size());
+        }
+
+        assertSampleNamesMatchInputVCF(originalHeader.getSampleNamesInOrder(), sampleNamesOriginalOrdering);
+        final List<String> batchSortedSampleNames = getBatchSortedList(sampleNamesOriginalOrdering, batchSize);
+
+        final VCFHeader remappedHeader = new VCFHeader(newHeaderLines, batchSortedSampleNames);
+        logger.info("Writing the new header with corrected sample names");
+        writer = createVCFWriter(output);
+        writer.writeHeader(remappedHeader);
+        logger.info("Copying the rest of the VCF");
+    }
+
+    private static void assertSampleNamesMatchInputVCF(final ArrayList<String> sampleNamesFromVCF, final List<String> sampleNameFromSampleMapFile) {
+        if(!CollectionUtils.isEqualCollection(sampleNamesFromVCF, sampleNameFromSampleMapFile)){
+            throw new UserException("The sample names in the provided sample name map file do not match the sample names in the provided vcf.\n" +
+                    "It is important this tool is run with exactly the same sample name map file that was used to create the vcf.");
+        }
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        writer.add(variant);
+    }
+
+    @Override
+    public void closeTool() {
+        logger.info("Finished writing the new vcf with corrected sampled names.");
+        if( writer != null) {
+            writer.close();
+        }
+    }
+    private void assertThatTheyReallyWantToProceed(){
+        if( !skipCommandLinePrompt ){
+            final Scanner scanner = new Scanner(System.in);
```

Comment:
Wrap `Scanner` construction in a try-with-resources so that it will be closed.

Reply:
done